### PR TITLE
feat(mechanics): Mission "requires" allows standard Player Conditions

### PR DESCRIPTION
**Pull request by SomeTroglodyte**
**Feature**
(With minor **Content** tweaks requiring the code)

## Summary
Today while playing Hai Reveal I ran into the irksome ["Sayari won't accept Nanachi's pendant" message](https://github.com/endless-sky/endless-sky/blob/ab9a3dd4fc0e1e27f74f085af35a984dc0ac00f0/data/hai/hai%20reveal%204%20middle.txt#L1844), and subsequently fixed it.

Notes
* A comment in that mission indicates the mission author intended to allow other keystones but couldn't.
* The proposed mission change allows _any_ keystone including plugin ones, the alternative to allow only the ones "Hai Pilgrimage" recognizes has some merit too - doable using an `or` condition.
* Related to [#10191](https://github.com/endless-sky/endless-sky/issues/10191) and other issues I can't find at the moment - on one side the capability to check attributes in "to ..." branches already exists but is underused in content, on the other side this PR extends the capabilities of "on ..." branches.
* `mission "Remnant: Expanded Horizons Astral 2"` should perhaps check `"hyperjumps to system: <name>"` instead, in the unlikely case a plugin has jump drive with a *shorter* jump range that's not sufficient to reach SagA - but since that one states it's not wormhole-capable I refrained from testing that.

## Usage examples
Now
```
	on complete
		require
			has "outfit (flagship installed): Quantum Keystone"
```
is a valid alternative to
```
	on complete
		require "Quantum Keystone"
```
... But being full player conditions this now allows complex tests.

## Testing Done
Most of these changes have been in my codebase and thus been playtested extensively for months, additionally I tested this specific mission explicitly, checking that it now allows any key stone to succeed while still failing when I uninstall all pendant, skeleton and normal variants.

## Save File
Faked with the mighty xed, but enough to depart, land, get the mission (so it's copied from data and not included in my save), and play it - the Pendant is installed, the others are in storage:
[Sayari Meep~3013-11-16.txt.zip](https://github.com/user-attachments/files/17020173/Sayari.Meep.3013-11-16.txt.zip)

## Wiki Update
Yes, TBD - if the idea pleases. As I'm expecting objections I'll postpone the effort.

## Performance Impact
N/A

### CODE REVIEW

File: source/MissionAction.cpp [hunk 1]
```diff
@@ -168,6 +168,10 @@ void MissionAction::LoadSingle(const DataNode &child)
 		else
 			child.PrintTrace("Error: Skipping invalid \"require\" count:");
 	}
+	else if(key == "require" && !hasValue && child.HasChildren())
+	{
+		requiredConditions.Load(child);
+	}
```
Comment (TomGoodIdea) [inline source/MissionAction.cpp:174]: You can merge this with the above `if`:
```
...
else if(key == "require")
{
	if(hasValue)
		require outfits...
	else if(child.HasChildren())
		require conditions...
}
```

Comment (TomGoodIdea) [reply source/MissionAction.cpp:174]: I still don't like it, but I guess it's acceptable then

File: data/coalition/coalition missions.txt [hunk 2]
```diff
@@ -160,7 +160,8 @@ mission "lost in coalition"
 	to offer
 		has "license: Coalition"
 	on offer
-		require "Jump Drive" 0
+		require
+			not "flagship attribute: jump drive"
```
Comment (Amazinite) [inline data/coalition/coalition missions.txt:164]: I believe these are older missions that predate this "flagship attribute" autocondition. The `require` node has been effectively replaced with the ability to use autoconditions that check your outfits in the `to *` node. What mechanically do we actually gain by turning the `require` node into effectively another condition set check?

**Review by Amazinite — CHANGES_REQUESTED** [review]
This seems unnecessary to me. Unless there's some mechanic that this provides us that can't be achieved currently (and I'm not able to think of any such thing), we should be replacing existing uses of the `require` node with autoconditions, not enhancing the node to do more than it does now.

File: data/coalition/coalition missions.txt (hunk 2 above)
Comment (Amazinite) [reply data/coalition/coalition missions.txt:164]: Yes. We have `to complete`, `to offer`, `to fail`, and `to accept`.

**Pull request closed without merging**